tests/RPM/Parse_parseRPMSpec.hs

```diff
@@ -0,0 +1,113 @@
+
+            -- section payload (112 bytes in this example)
+            -- DSAHeader
+            0x88, 0x46, 0x04, 0x00, 0x11, 0x02, 0x00, 0x06, 0x05, 0x02, 0x53, 0x67, 0xab, 0x9d, 0x00, 0x0a,
+            0x09, 0x10, 0x50, 0x8c, 0xe5, 0xe6, 0x66, 0x53, 0x4c, 0x2b, 0x6b, 0x83, 0x00, 0xa0, 0x9e, 0x1c,
+            0x4e, 0x19, 0xd5, 0x78, 0x37, 0x53, 0x61, 0x8a, 0x34, 0x4b, 0x40, 0x91, 0xfb, 0xc1, 0x24, 0xbb,
+            0x1c, 0x62, 0x00, 0x9f, 0x50, 0xe6, 0x5c, 0x34, 0x72, 0xa6, 0x54, 0x70, 0x45, 0xce, 0xe9, 0xec,
+            0x02, 0x6b, 0x98, 0xfa, 0x45, 0x72, 0x8f, 0xca, -- SHA1Header
+                                                            0x66, 0x36, 0x37, 0x35, 0x64, 0x37, 0x39, 0x62,
+            0x66, 0x66, 0x33, 0x34, 0x34, 0x66, 0x36, 0x63, 0x63, 0x63, 0x32, 0x64, 0x34, 0x65, 0x37, 0x31,
+            0x66, 0x66, 0x62, 0x62, 0x38, 0x61, 0x39, 0x63, 0x36, 0x38, 0x39, 0x62, 0x61, 0x64, 0x65, 0x63,
+            -- no signature padding here
+
+            -- begin header section
+            0x8e, 0xad, 0xe8, 0x01, 0x00, 0x00, 0x00, 0x00, -- section signature (3b), sectionVersion (1b), reserved (4b)
+            0x00, 0x00, 0x00, 0x03, -- sectionCount (4b) -- MODIFIED FOR BREVITY
+            0x00, 0x00, 0x00, 0x36, -- sectionSize (4b) -- MODIFIED FOR BREVITY
+
+            -- tags defined in this section, sectionCount * 16 bytes (32 bytes in this example)
+            -- 1000 6 2 1 == Name (string)
+            0x00, 0x00, 0x03, 0xe8, 0x00, 0x00, 0x00, 0x06,  0x00, 0x00, 0x00, 0x02, 0x00, 0x00, 0x00, 0x01,
+            -- 1001 6 6 1 == Version (string)
+            0x00, 0x00, 0x03, 0xe9, 0x00, 0x00, 0x00, 0x06,  0x00, 0x00, 0x00, 0x06, 0x00, 0x00, 0x00, 0x01,
+            -- 1004 6 15 1 == Summary (i18n string)
+            0x00, 0x00, 0x03, 0xec, 0x00, 0x00, 0x00, 0x09,  0x00, 0x00, 0x00, 0x0f, 0x00, 0x00, 0x00, 0x01,
+
+            -- section payload (54 bytes in this example)
+            0x43, 0x00, -- Name
+                        0x76, 0x6c, 0x63, 0x00, -- Version
+                                                0x32, 0x2e,  0x31, 0x2e, 0x34, 0x00,
+                                                                                     0x31, 0x34, 0x00, -- Summary
+                                                                                                       0x41,
+            0x20, 0x66, 0x72, 0x65, 0x65, 0x20, 0x61, 0x6e,  0x64, 0x20, 0x63, 0x72, 0x6f, 0x73, 0x73, 0x2d,
+            0x70, 0x6c, 0x61, 0x74, 0x66, 0x6f, 0x72, 0x6d,  0x20, 0x6d, 0x65, 0x64, 0x69, 0x61, 0x20, 0x70,
+            0x6c, 0x61, 0x79, 0x65, 0x72, 0x00, -- RPM payload -- MODIFIED FOR BREVITY
+                                                0x54, 0x45,  0x53, 0x54, 0x2D, 0x54, 0x45, 0x53, 0x54, 0x21]
+      -- parsing succeeds
+      parseRPM `shouldSucceedOn` stream
+      -- no unconsumed input
+-- TODO: I am not sure why this fails ATM, it says parser didn't leave anything behind
+--      stream ~?> parseRPM `leavesUnconsumed` BS.pack []
```

Comment:
@clumens however now I have a new problem. This line fails with:
```
  src/Test/Hspec/Attoparsec.hs:129: 
  1) RPM.Parse_parseRPM.RPM.Parse.parseRPM succeeds with valid data
         expected the parser to leave the following unconsumed: ""
         but got no unconsumed input
```
From what I can tell the parser leaves `Nothing` unconsumed hence the failure. What I'm still trying to figure out is why I don't get the same failure at all the other places I use a similar assertion.